components/authz-service/storage/postgres/migration/sql/52_query_returns_condition_type.up.sql

```diff
@@ -0,0 +1,194 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION
+  query_staged_and_applied_rules(_project_filter TEXT[])
+  RETURNS setof json AS $$
+
+  -- fetch staged rules
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', r.project_id,
+    'name', r.name,
+    'type', r.type,
+    'deleted', r.deleted,
+    'status', 'staged',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute,
+        -- condition type always matches the rule type
+        'type', r.type
```

Comment:
❓ We could also change this in our Go code, couldn't we? If the conditions' types always match the rule type, it's weird that they've got such a field at all. What am I missing? 😃

Reply:
We use the condition type to enforce what attributes can be set: https://github.com/chef/automate/blob/master/components/authz-service/storage/v2/condition.go#L63

I agree though, if there's another way to enforce that, I'd prefer not to have a type on conditions

Reply:
...but isn't that only for _creating new conditions_? I'm still not sure I see the point of _returning it_ from the database query `1+n` times, once in the rule object, and once in every single one of its conditions?

Reply:
@srenatus what do you think about https://github.com/chef/automate/pull/738/commits/a9d02449e127977d3aad3d2a7b06290649599ff0?

Reply:
:100: much less weird. Thanks for doing the no-pain-no-gain kind of work there 👍